lib/src/rules/cascade_invocations.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.cascade_invocations;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/linter.dart';
+
+const _desc = r'Cascade consecutive method invocations on the same reference.';
+
+const _details = r'''
+
+**DO** Use the cascading style when succesively invoking methods on the same
+ reference.
+
+**BAD:**
+```
+SomeClass someReference = new SomeClass();
+someReference.firstMethod();
+someReference.secondMethod();
+```
+
+**BAD:**
+```
+SomeClass someReference = new SomeClass();
+...
+someReference.firstMethod();
+someReference.secondMethod();
+```
+
+**GOOD:**
+```
+SomeClass someReference = new SomeClass()
+    ..firstMethod()
+    ..secondMethod();
+```
+
+**GOOD:**
+```
+SomeClass someReference = new SomeClass();
+...
+someReference
+    ..firstMethod()
+    ..secondMethod();
+```
+
+''';
+
+class CascadeInvocations extends LintRule {
+  _Visitor _visitor;
+  CascadeInvocations()
+      : super(
+            name: 'cascade_invocations',
+            description: _desc,
+            details: _details,
+            group: Group.style) {
+    _visitor = new _Visitor(this);
+  }
+
+  @override
+  AstVisitor getVisitor() => _visitor;
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+  _Visitor(this.rule);
+
+  @override
+  void visitMethodInvocation(MethodInvocation node) {
+    if (node.parent is! ExpressionStatement ||
+        node.realTarget is! SimpleIdentifier) {
+      return;
+    }
+
+    final simpleIdentifier = node.realTarget as SimpleIdentifier;
+    Block body = node.parent.parent;
+    Iterable previousChilds =
+        body.childEntities.takeWhile((n) => n != node.parent);
+    if (_previousIsMethodInvocation(previousChilds)) {
+      _reportIfConsecutiveInvocationsOnSameTarget(
+          previousChilds, simpleIdentifier, node);
+    }
+
+    if (previousChilds.last is VariableDeclarationStatement) {
+      _reportIfDeclarationFollowedByMethodInvocation(
+          previousChilds, simpleIdentifier, node);
+    }
+  }
+
+  bool _previousIsMethodInvocation(Iterable previousChilds) =>
+      previousChilds.isNotEmpty &&
+      previousChilds.last is ExpressionStatement &&
+      previousChilds.last.expression is MethodInvocation;
+
+  bool _previousIsSingleVariableDeclaration(
+          Iterable previousChilds, VariableDeclarationStatement variables) =>
+      previousChilds.isNotEmpty &&
```

Comment:
I don't understand why we need to test that the collection is not empty. And I don't think that we need to test that `variables` is not null. If we can remove both tests, then this method could be inlined.